sqlmesh/core/test/definition.py

```diff
@@ -379,8 +380,14 @@ def generate_test(
             f"Fixture '{fixture_path}' already exists, make sure to set --overwrite if it can be safely overwritten."
         )
 
+    # ruamel.yaml does not support pandas Timestamps, so we must convert them to python
+    # datetime or datetime.date objects based on column type
     inputs = {
-        dep: engine_adapter.fetchdf(query).to_dict(orient="records")
+        models[dep]
+        .name: pandas_timestamp_to_pydatetime(
```

Comment:
Have you double-checked the use of `name` vs `fqn` in this section? E.g. should this be `models[dep].fqn`?